File: api/src/main/java/com/mystore/manager/api/admin/criteria/ApiKeyCriteria.java
package com.mystore.manager.api.admin.criteria;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.mystore.manager.api.common.criteria.PaginationCriteria;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@JsonIgnoreProperties(ignoreUnknown = true)
public class ApiKeyCriteria extends PaginationCriteria {
    private Integer id;
    private Integer posId;
    private String description;
    private Boolean active;
}

File: ui-admin/src/app/backend/criteria/admin/api-key-criteria.ts
import { CommonCriteria } from '../commoncriteria';

export class ApiKeyCriteria extends CommonCriteria {
    id: number;
    posId: number;
    description: string;
    active: boolean;

    constructor() {
        super();
        this.id = null;
        this.posId = null;
        this.description = null;
        this.active = null;
    }
}
